### ISSUE TITLE
Error creating google_monitoring_uptime_check_config with private Service Directory service

### ISSUE THREAD

**Issue by mikepietruszka** [hashicorp/terraform-provider-google#11537]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
```
$ terraform -v
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.17.0
+ provider registry.terraform.io/hashicorp/google-beta v4.17.0
```

### Affected Resource(s)

* google_monitoring_uptime_check_config

### Terraform Configuration Files

```tf
resource "google_monitoring_uptime_check_config" "http" {
  display_name = "http-uptime-check"
  timeout = "60s"
  project  = "test_project_name"

  http_check {
    path = "/web"
    port = "80"
    request_method = "POST"
    content_type = "URL_ENCODED"
    body = "ZmXXXXXXXXXI="
  }

  monitored_resource {
    type = "servicedirectory_service"
    labels = {
      project_id = "test_project_name"
      location = "us-central1"
      namespace_name = "custom_namespace"
      service_name = "web"
    }
  }
}
```

### Debug Output

```
google_monitoring_uptime_check_config.http: Creating...
╷
│ Error: Error creating UptimeCheckConfig: googleapi: Error 400: Configs with Service Directry targets must use checker_type VPC_CHECKERS
│ 
│   with google_monitoring_uptime_check_config.http,
│   on main.tf line 41, in resource "google_monitoring_uptime_check_config" "http":
│   41: resource "google_monitoring_uptime_check_config" "http" {
│ 
```

### Expected Behavior

Private Uptime Check should have been created with Service Directory target. This typically works using `gcloud` command.

### Actual Behavior

Private Uptime Check fails to create because the schema in [`resourceMonitoringUptimeCheckConfig`](https://github.com/hashicorp/terraform-provider-google/blob/791b9ef6d986c585634a52b3ba46e89fc34db5a6/google/resource_monitoring_uptime_check_config.go#L48) is missing `checker_type`/`CheckerType` property that is defined in [`Resource: UptimeCheckConfig`](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.uptimeCheckConfigs#resource:-uptimecheckconfig) object.

[`CheckerType`](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.uptimeCheckConfigs#checkertype) states that `VPC_CHECKERS` should be used for uptime checks using Service Directory targets.

### Steps to Reproduce

1. Define `main.tf` with `google_monitoring_uptime_check_config` resource.
2. Run `terraform apply`.

### References

* #0000

**Comment by shuyama1** [comment]
Hi @mikepietruszka! Thanks for filing the issue. Looks like the provider does not currently have `checker_type` implemented. Therefore change the label of this issue from `bug` to `enhancement`.

Note: API for `checkerType`: https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.uptimeCheckConfigs#CheckerType

**Comment by mikepietruszka** [comment]
Hi @shuyama1, I have a working fixed (that's been tested) in PR #11545. I attempted to do this via Magic Modules but was having issues getting it setup on my machine.

**Comment by jrbarnard** [comment]
I created a PR in magic modules to add this field - https://github.com/GoogleCloudPlatform/magic-modules/pull/5987
